packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts

```diff
@@ -0,0 +1,352 @@
+};
+
+export async function scheduleWalkthrough(context: any, params: Partial<FunctionParameters>): Promise<Partial<FunctionParameters>> {
+  const projectBackendDirPath = context.amplify.pathManager.getBackendDirPath();
+  const resourceDirPath = path.join(projectBackendDirPath, categoryName, params.resourceName);
+  const cfnFileName = `${params.resourceName}-cloudformation-template.json`;
+  const cfnFilePath = path.join(resourceDirPath, cfnFileName);
+  let scheduleParams: Partial<FunctionParameters> = params;
+  if (params.cloudwatchEnabled === undefined || params.cloudwatchEnabled === 'false') {
+    if (await context.amplify.confirmPrompt.run('Do you want this function to be invoked on a schedule?', false)) {
+      scheduleParams.cloudwatchEnabled = 'true';
+      try {
+        let cloudWatchRule = await cronServiceWalkthrough(context);
+        scheduleParams.cloudwatchRule = cloudWatchRule;
+        if (context.input.command === 'update') {
+          //append cloudwatch events to CFN File
+          const cfnContent = context.amplify.readJsonFile(cfnFilePath);
+          cfnContent.Resources.CloudWatchEvent = {
+            Type: 'AWS::Events::Rule',
+            Properties: {
+              Description: 'Schedule rule for Lambda',
+              ScheduleExpression: {
+                Ref: 'CloudWatchRule',
+              },
+              State: 'ENABLED',
+              Targets: [
+                {
+                  Arn: { 'Fn::GetAtt': ['LambdaFunction', 'Arn'] },
+                  Id: {
+                    Ref: 'LambdaFunction',
+                  },
+                },
+              ],
+            },
+          };
+          // append permissions to invoke lambda via cloiudwatch to CFN file
+          cfnContent.Resources.PermissionForEventsToInvokeLambda = {
+            Type: 'AWS::Lambda::Permission',
+            Properties: {
+              FunctionName: {
+                Ref: 'LambdaFunction',
+              },
+              Action: 'lambda:InvokeFunction',
+              Principal: 'events.amazonaws.com',
+              SourceArn: { 'Fn::GetAtt': ['CloudWatchEvent', 'Arn'] },
+            },
+          };
+          // append the outputs section of cloudwatchRULE
+          cfnContent.Outputs.CloudWatchEventRule = {
+            Value: {
+              Ref: 'CloudWatchEvent',
+            },
+          };
+          fs.writeFileSync(cfnFilePath, JSON.stringify(cfnContent, null, 4));
+        }
+      } catch (e) {
+        context.print.error(e.message);
+      }
+    }
+  } else {
+    if (await context.amplify.confirmPrompt.run('Do you want to update or remove the function schedule?', false)) {
+      const cfnContent = context.amplify.readJsonFile(cfnFilePath);
+      const scheduleEventOperationQuestion = {
+        type: 'list',
+        name: 'ScheduleEventOperation',
+        message: 'Select from the following options (Use arrow keys)',
+        choices: ['Update the schedule', 'Remove the schedule'],
```

Comment:
set a value for the choice so we don't have code logic depending on the display name of the option

Reply:
done

Reply:
I mean set the choices to:
```javascript
choices: [
  {
    name: 'Update the schedule',
    value: 'update',
  },
  {
    name: 'Remove the schedule',
    value: 'remove',
  }
]
```
Then use the values in the switch statement below. That way the switch statement conditions are not dependent on the wording of the selection but rather an internal value.